tests/basic/storage_pool_test.go

```diff
@@ -9993,3 +9994,283 @@ func exitPoolMaintenance(poolId string) {
 	}
 
 }
+
+func findNodeForReplAdd(vol *volume.Volume) (*node.Node, error) {
+	poolUUIDs, err := GetPoolIDsFromVolName(vol.ID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get pool uuids for volume %v: %v", vol.ID, err)
+	}
+
+	// list of nodes
+	nodes := make([]*node.Node, 0)
+	for _, p := range poolUUIDs {
+		n, err := GetNodeFromPoolUUID(p)
+		if err != nil {
+			return nil, fmt.Errorf("failed to find node for pool UUID %v: %v", p, err)
+		}
+		nodes = append(nodes, n)
+	}
+
+	// find node to repl add
+	for _, new := range node.GetStorageNodes() {
+		found := false
+		for _, old := range nodes {
+			if new.Id == old.Id {
+				found = true
+				break
+			}
+		}
+		if !found {
+			return &new, nil
+		}
+	}
+	return nil, fmt.Errorf("failed to find a node for repl add for volume %v", vol.ID)
+}
+
+var _ = Describe("{PoolDeleteFunctionality}", func() {
+	/*
+		Migrated from px-test: PoolDeleteFunctionality
+			1. Delete pools till total availble pool is 1.
+			2. Verify after each delete if the right pool got deleted.
+			3. Verify Alerts.
+			4. Try to delete the last pool in the node and it should fail.
+			5. Randomly pick a drive which got freeed up because of pool delete and add it back.
+	*/
+
+	JustBeforeEach(func() {
+		StartTorpedoTest("PoolDelete", "Initiate pool deletion", nil, 0)
+	})
+	var contexts []*scheduler.Context
+
+	ItLog := "Initiate pool delete, then add a new pool and expand the pool"
+	It(ItLog, func() {
+		selectedNode := node.GetStorageNodes()[0]
+		nodePools := selectedNode.StoragePools
+
+		poolToAddBack := nodePools[rand.Intn(len(nodePools))]
+		for _, pool := range nodePools {
+			poolID := strconv.Itoa(int(pool.ID))
+			deletePoolAndValidate(selectedNode, poolID)
+		}
+
+		// verify all pools deleted
+		pools, err := Inst().V.ListStoragePools(metav1.LabelSelector{})
+		log.FailOnError(err, "Failed to list storage pools")
+		dash.VerifyFatal(len(pools) == 0, true, "verify all pools deleted")
+
+		contexts = make([]*scheduler.Context, 0)
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			contexts = append(contexts, ScheduleApplications(fmt.Sprintf("pooldeletefunc-%d", i))...)
+		}
+		ValidateApplications(contexts)
+		defer appsValidateAndDestroy(contexts)
+
+		stepLog = fmt.Sprintf("Adding cloud drive to node [%v] with size [%v]", selectedNode.Name, poolToAddBack.TotalSize/units.GiB)
```

Comment:
could you print each step in the test?  which step is it in comment L10033~10037?